### PR TITLE
using mute variables in combinat

### PR DESCRIPTION
**Pull request by fchapoton**
taking care of unused variables found by `pylint` in combinat

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/combinat/sloane_functions.py [hunk 1]
```diff
@@ -600,10 +600,10 @@ def _eval(self, n):
...
             [1, 1, 2, 2, 3, 4, 5, 6, 7, 8, 11, 12, 15, 16]
         """
-        from sage.rings.big_oh import O
-        R, x = QQ[['x']].objgen()
-        p = 1/((1-x)*(1-x**2)*(1-x**5)*(1-x**10)+O(x**(n+4)))
-        return ZZ(p.coefficients()[n])
+        from sage.rings.lazy_series_ring import LazyPowerSeriesRing
+        x = LazyPowerSeriesRing(ZZ, 'x').gen()
+        p = 1/((1-x)*(1-x**2)*(1-x**5)*(1-x**10))
+        return ZZ(p.coefficient(n)
```
Comment (mkoeppe) [inline src/sage/combinat/sloane_functions.py]: unbalanced ()

File: src/sage/combinat/sloane_functions.py [hunk 2]
```diff
@@ -603,7 +603,7 @@ def _eval(self, n):
         from sage.rings.lazy_series_ring import LazyPowerSeriesRing
         x = LazyPowerSeriesRing(ZZ, 'x').gen()
         p = 1/((1-x)*(1-x**2)*(1-x**5)*(1-x**10))
-        return ZZ(p.coefficient(n)
+        return ZZ(p.coefficient(n))
```
Comment (mantepse) [inline src/sage/combinat/sloane_functions.py]: Just `p[n]` would be better.  But using a lazy generating function to return a coefficient is a bit odd, why not do `return Partitions(n, parts_in=[1,2,5,10]).cardinality()`, this is much faster and also clearer.

File: src/sage/combinat/sloane_functions.py (hunk 1 above)
Comment (mantepse) [reply src/sage/combinat/sloane_functions.py]: Using `LazyPowerSeries` here is an extremely bad idea!  Getting the coefficient of 100000 is likely to take forever.

File: src/sage/combinat/sloane_functions.py (hunk 2 above)
Comment (fchapoton) [reply src/sage/combinat/sloane_functions.py]: done using Partitions, but this seems to be dead slow...

Comment (mantepse) [reply src/sage/combinat/sloane_functions.py]: That's strange, can you show a comparison?  On my computer, using Partitions is much faster:
```
sage: a = sloane.A000008
sage: %timeit a[100000]
48.3 ms ± 2.25 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
sage: %timeit Partitions(100000, parts_in=[1,2,5,10]).cardinality()
13.5 ms ± 198 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Comment (fchapoton) [reply src/sage/combinat/sloane_functions.py]: Maybe it was the internal launch of gap that made it slow

File: src/sage/combinat/yang_baxter_graph.py [hunk 3]
```diff
@@ -546,7 +546,7 @@ def relabel_edges(self, edge_dict, inplace=True):
             from copy import copy
             Y = copy(self)
         digraph = Y._digraph
-        for u, v, i in digraph.edges(sort=True):
+        for u, v in digraph.edges(sort=True, labels=False):
```
Comment (dcoudert) [inline src/sage/combinat/yang_baxter_graph.py]: `sort=True` is not needed here.

**Review by dcoudert — CHANGES_REQUESTED** [review]
otherwise, LGTM.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by fchapoton** [comment]
Merci. Je peux passer en positif, alors ?

File: src/sage/combinat/sloane_functions.py (hunk 2 above)
Comment (dcoudert) [reply src/sage/combinat/sloane_functions.py]: on macOS M1, I have similar behavior than @mantepse 
```py
sage: a = sloane.A000008
sage: %timeit a[100000]
26.1 ms ± 262 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
sage: %timeit Partitions(100000, parts_in=[1,2,5,10]).cardinality()
8.51 ms ± 309 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

**Pull request merged**